mountpoint-s3/src/build_info.rs

```diff
@@ -9,18 +9,39 @@ mod built {
 
 /// Valid SemVer version constructed using declared Cargo version and short commit hash if needed.
 pub const FULL_VERSION: &str = {
+    const BUILD_TARGET_STR: &str = match build_target_suffix() {
+        Some(target) => target,
+        None => "",
+    };
+
     if is_official_aws_release() {
-        built::PKG_VERSION
+        if build_target_suffix().is_some() {
+            const_format::concatcp!(built::PKG_VERSION, "+", BUILD_TARGET_STR)
+        } else {
+            built::PKG_VERSION
+        }
     } else {
         const_format::concatcp!(built::PKG_VERSION, "-unofficial", git_commit_suffix())
     }
 };
-
-/// Checks environment to see if this build is for an official Mountpoint for Amazon S3 release.
 const fn is_official_aws_release() -> bool {
     option_env!("MOUNTPOINT_S3_AWS_RELEASE").is_some()
 }
 
+/// Gets build target from BUILD_TARGET environment variable if set
+const fn build_target_suffix() -> Option<&'static str> {
+    match option_env!("MOUNTPOINT_S3_AWS_RELEASE") {
```

Comment:
Try to avoid duplicating the env variable name.

Reply:
Also, are we keeping the original `MOUNTPOINT_S3_AWS_RELEASE`? Or should we rather use something like  `MOUNTPOINT_S3_AWS_RELEASE_TARGET`?